### PR TITLE
Fix cs_set_user_team

### PR DESCRIPTION
**Pull request by Sylvert0**
When the player change of team with CS_DONTCHANGE its change anyway! 
Plugin to test: http://pastebin.com/HCixm6UL

### CODE REVIEW

File: plugins/include/cstrike.inc [hunk 1]
```diff
@@ -32,6 +32,7 @@
  */
 enum CsInternalModel
 {
+	CS_AUTOCHANGE = -1
```
Comment (Arkshine) [inline plugins/include/cstrike.inc]: Come one, at least try to compile before doing a PR.

**Comment by Arkshine** [comment]
This breaks backward compatibility as it is. Old compiled plugins which have `CS_DONTCHANGE` set, should expect `ResetModel()` be called by default. 

Same for default value in native header, this will change behavior for existing plugin relying on default value, coder should not have to modify their plugin.

Though, this is a specific situation where you set model before team at the same frame, this can be checked, and for sure this doesn't make sense to reset model if you set before a new one.

I see three ways:
* You change the default value to `CS_AUTOCHANGE`, but then, this would require more checks to match only context of having setting model right before team.  Basically resetting should happen only if you pass `CS_DONTCHANGE` and above, or player doesn't have custom model, or player haven't just got a custom model (not applied yet).
* You rename `CS_AUTOCHANGE` to something like maybe `CS_NORESET`, then all you need to do change is to allow reset with any values `>= 0`. 
* We could argue that this can be just documented that setting a model should be done after setting a team. 

First will require some more code to make sure it fixes only this issue. I think it needs to track when `UpdateModel` is invoked. 
Second is easier and safer, there is no bcompat issues, and we still give a way to fix such situation.
Third is the lazy way but I don't see any reasons not to provide a way to deal with this behavior.

I don't have preferences for now, just posting some thoughts.

**Comment by Arkshine** [comment]
Using the second way is fine.

For doc, I would use more this for consistency: "or if CS_NORESET the game will not update it".

By the way, you should avoid editing directly on github, but instead working on a branch on your computer, then pushing this branch to create a PR. This way, if you need to modify your commits you can do it easily. Some help https://wiki.alliedmods.net/Git_Tutorial#Contributing_via_Forks. You have also software like https://desktop.github.com/ or http://www.syntevo.com/smartgit/.

Anyway, you have 7 commits for small changes, please squash them to one commit.(some help: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)

**Comment by Arkshine** [comment]
Can you squash your commits as said above, please?

**Comment by Sylvert0** [comment]
I delete for mistake the repository trying to squash the commits xD So i forked again the repository and make a new branch https://github.com/Sylvert0/amxmodx/tree/Fix-cs_set_user_team you can close this?

**Comment by Arkshine** [comment]
I'm not sure how you did that by mistake as to delete a repo you need to go in options and confirming it. If you wanted some help you could have asked. But well, okay.

**Pull request closed without merging**